### PR TITLE
Add Advanced Fusion Reactor tech and item.

### PR DESCRIPTION
**Pull request by Xenega**
Add Advanced Fusion Reactor tech and item.

### CODE REVIEW

File: data/items.dat [hunk 1]
```diff
@@ -88,10 +88,17 @@ build_list = ALL
 [Fusion Reactor]
 cost_list = 50000 | 0 | 9
 type = reactor
-quality_list = discover_modifier | 2500
+quality_list = discover_modifier | 2200
 pre = Fusion Reactor
```
Comment (rofl0r) [inline data/items.dat]: why does this have itself as prereq ?

Comment (Xenega) [reply data/items.dat]: No, Fusion Reactor is a tech. And a item.

Comment (Xenega) [reply data/items.dat]: To clarify, Fusion Reactor the item depends of Fusion Reactor the tech.

Comment (sunfall) [reply data/items.dat]: Maybe we then want to make the tech 'Fusion' and 'Advanced Fusion' to make this less confusing?  (Not required for this change, just food for thought.)

File: data/techs_str.dat [hunk 2]
```diff
@@ -59,6 +59,11 @@ name = Fusion Reactor
 description = Modern fusion technology is woefully inadequate for my purposes.  With my research abilities, I should be able to bypass the roadblocks of modern human scientists and come up with a small, efficient fusion reactor.
 result = I can now construct Fusion Reactors for my bases.
 
+[Advanced Fusion Reactor]
+name = Advanced Fusion Reactor
+description = By pushing the energy of fusion at his highest level, I should come with more efficiently model for my Fusion Reactor.
```
Comment (sunfall) [inline data/techs_str.dat]: My Fusion Reactors have unexpected fluctuations in their energy output at high levels of utilization.  Further research should result in a more efficient design for my Reactors.

**Pull request merged**